homeassistant/components/remote/__init__.py

```diff
@@ -110,3 +125,15 @@ def async_send_command(self, command, **kwargs):
         """
         return self.hass.async_add_job(ft.partial(
             self.send_command, command, **kwargs))
+
+    def learn_command(self, **kwargs):
+        """Learn a command from a device."""
+        raise NotImplementedError()
+
+    def async_learn_command(self, **kwargs):
```

Comment:
This should be made a coroutine, using `async def`.

Reply:
no, because we return a coroutine/task object

Reply:
No we shouldn't use that pattern anymore. We should define coroutines and await the task inside them.

Reply:
Done.